dapps/marketplace/src/components/BottomScrollListener.js

```diff
@@ -65,19 +65,40 @@ const BottomScrollListener = ({
     }
   }, rebindListenerOnProps)
 
-  window.requestAnimationFrame(() => {
-    if (
-      ((bindOnContainer &&
-        elementRef.current.scrollHeight < elementRef.current.clientHeight) ||
-        (!bindOnContainer &&
-          document.body.clientHeight < window.innerHeight)) &&
-      ready &&
-      hasMore
-    ) {
+  const onWindowResize = useCallback(() => {
+    if (!ready || !hasMore) {
+      return
+    }
+
+    let canFetchMore = false
+
+    if (!bindOnContainer && document.body.clientHeight <= window.innerHeight) {
+      canFetchMore = true
+    }
+
+    const hasMoreSpace =
+      elementRef.current.clientHeight <=
+      elementRef.current.parentElement.clientHeight
+
+    if (bindOnContainer && hasMoreSpace) {
+      canFetchMore = true
+    }
+
+    if (canFetchMore) {
       onBottom()
     }
   })
 
+  useEffect(() => {
+    const debounced = debounce(onWindowResize, 300)
```

Comment:
minor - would be great to have the `300ms` as a constant.

Reply:
Can be done. I don't think we have a constants file for DApp. So, Not prioritising this now. Will comeback to it later.